sechub-test/src/test/java/com/mercedesbenz/sechub/test/encryption/EncryptionUsageTest.java

```diff
@@ -0,0 +1,165 @@
+
+        EncryptionRotator rotatorToTest = new EncryptionRotator();
+
+        /* test */
+
+        /* @formatter:off */
+        EncryptionRotationSetup rotateSetup = EncryptionRotationSetup.builder().
+                newCipher(cipherNewPassword).
+                oldCipher(cipherOldPassword).
+                oldInitialVector(encryptionResult.getInitialVector()).
+        build();
+
+        byte[] newEncrypted = rotatorToTest.rotate(encryptionResult.getEncryptedData(), rotateSetup);
+
+        /* test */
+
+        /* @formatter:on */
+        String unencryptedTestData = encryptSupport.decryptString(newEncrypted, cipherNewPassword, rotateSetup.getNewInitialVector());
+
+        assertThat(testData).isEqualTo(unencryptedTestData);
+
+    }
+
+    @Test
+    void rotation_with_changed_password_different_init_vector() {
+        /* prepare */
+        String testData = "hallo welt";
+        byte[] oldPassword = "x".repeat(32).getBytes();
+        byte[] newPassword = "y".repeat(32).getBytes();
+
+        PersistentCipherFactory factory = new PersistentCipherFactory();
+        PersistentCipherType cipherType = PersistentCipherType.AES_GCM_SIV_256;
+        PersistentCipher cipherOldPassword = factory.createCipher(new DefaultSecretKeyProvider(oldPassword, cipherType), cipherType);
+
+        EncryptionSupport encryptSupport = new EncryptionSupport();
+
+        // encrypt with old cipher
+        EncryptionResult firstEncryptionResult = encryptSupport.encryptString(testData, cipherOldPassword);
+
+        PersistentCipher cipherNewPassword = factory.createCipher(new DefaultSecretKeyProvider(newPassword, cipherType), cipherType);
+        InitializationVector newInitVector = cipherNewPassword.createNewInitializationVector();
+
+        EncryptionRotator rotatorToTest = new EncryptionRotator();
+
+        /* test */
+
+        /* @formatter:off */
+        EncryptionRotationSetup rotateSetup = EncryptionRotationSetup.builder().
+                newCipher(cipherNewPassword).
+                oldCipher(cipherOldPassword).
+                oldInitialVector(firstEncryptionResult.getInitialVector()).
+                newInitialVector(newInitVector).
+                build();
+
+        byte[] newEncrypted = rotatorToTest.rotate(firstEncryptionResult.getEncryptedData(), rotateSetup);
+
+        /* test */
+
+        /* @formatter:on */
+        String unencryptedTestData = encryptSupport.decryptString(newEncrypted, cipherNewPassword, newInitVector);
+
+        assertThat(testData).isEqualTo(unencryptedTestData);
+
+        // additional test - we check that the decryption with old init vector does not
+        // work
+        assertThatThrownBy(() -> encryptSupport.decryptString(newEncrypted, cipherNewPassword, firstEncryptionResult.getInitialVector()))
+                .isInstanceOf(IllegalStateException.class).hasRootCauseInstanceOf(AEADBadTagException.class);
+    }
```

Comment:
Just an idea for the future, we could also extend this tests later:
For example, we could add a test where the new cipher also uses a different `PersistentCipherType`, like `PersistentCipherType.AES_GCM_SIV_128` or something completely different as soon as we want to support something else.